doc/Technical Documentation/Compiler Intermediate Language format.md

```diff
@@ -11,124 +11,126 @@ languages it is more similar to assembly languages than high level languages.
 have been applied and registers have been assigned
 
 The initial representation looks something like this:
+```
+segment end exitseg
+
+segment codeseg
+
+global _main
+
+
+global _main
+_main:
+; Line 3: int main(int argc, char **argv) 
+; Line 4: { 
+
+
+  BLOCK 1
+  PROLOGUE    unset
+  BLOCK 2
+
+L_2:
+  DBG BLOCK START
+  DBG BLOCK START
+  DBG BLOCK START
+  _i:LINK(0).I =   #0.I
+  T0.I =   _i:LINK(0).I
+  CONDGO  L_7:PC ; T0.I S>= #A.I
+  BLOCK 3
+
+L_5:
+; Line 6:         printf("hi %d\n", i); 
+
+  T0.I =   _i:LINK(0).I
+  PARM    T0.I
+  PARM    #L_1:PC.A
+  TAG
+  GOSUB   #_printf:PC.UI
+  PARMADJ #8.N
+  DBG BLOCK END
+  BLOCK 4
+
+L_8:
+  T0.I =   _i:LINK(0).I
+  T2.I = T0.I + #1.I
+  _i:LINK(0).I =   T2.I
+; Line 5:     for (int i=0; i < 10; i++) 
+
+  BLOCK 5
 
-> segment end exitseg
-> 
-> segment codeseg
-> 
-> global _main
-> 
-> 
-> global _main
-> _main:
-> ; Line 3: int main(int argc, char \*\*argv) 
-> ; Line 4: { 
-> 
-> 
->   BLOCK 1
->   PROLOGUE    unset
->   BLOCK 2
-> 
-> L_2:
->   DBG BLOCK START
->   DBG BLOCK START
->   DBG BLOCK START
->   _i:LINK(0).I =   #0.I
->   T0.I =   _i:LINK(0).I
->   CONDGO  L_7:PC ; T0.I S>= #A.I
->   BLOCK 3
-> 
-> L_5:
-> ; Line 6:         printf("hi %d\n", i); 
-> 
->   T0.I =   _i:LINK(0).I
->   PARM    T0.I
->   PARM    #L_1:PC.A
->   TAG
->   GOSUB   #_printf:PC.UI
->   PARMADJ #8.N
->   DBG BLOCK END
->   BLOCK 4
-> 
-> L_8:
->   T0.I =   _i:LINK(0).I
->   T2.I = T0.I + #1.I
->   _i:LINK(0).I =   T2.I
-> ; Line 5:     for (int i=0; i < 10; i++) 
-> 
->   BLOCK 5
-> 
 L_6:
->   T0.I =   _i:LINK(0).I
->   CONDGO  L_5:PC ; T0.I S< #A.I
->   BLOCK 6
-> 
-> L_7:
->   DBG BLOCK END
-> ; Line 7: } 
-> 
->   DBG BLOCK END
->   RV.T3.I =   #0.I
->   GOTO    L_3:PC
->   BLOCK 7
-> 
-> L_3:
->   EPILOGUE    unset
->   RET #0.UI
-> 
-> 
-> segment end codeseg
->
-> segment stringseg
-> 
-> L_1:
-> 
->   DC.B "hi %d"
->   DC.B 0xa    
->   DC.B 0x0
->   extern _printf
->
+  T0.I =   _i:LINK(0).I
+  CONDGO  L_5:PC ; T0.I S< #A.I
+  BLOCK 6
 
-Here the Txxx variables are compiler-generated temporaries; they haven't been assigned to either registers or the stack at this point.
+L_7:
+  DBG BLOCK END
+; Line 7: } 
 
-Lines such as:
+  DBG BLOCK END
+  RV.T3.I =   #0.I
+  GOTO    L_3:PC
+  BLOCK 7
+
+L_3:
+  EPILOGUE    unset
+  RET #0.UI
+
+
+segment end codeseg
 
-> T0.I  = _i:LINK(0).I
+segment stringseg
 
+L_1:
+
+  DC.B "hi %d"
+  DC.B 0xa    
+  DC.B 0x0
+  extern _printf
+```
+
+Here the Txxx variables are compiler-generated temporaries; they haven't been assigned to either registers or the stack at this point.
+
+Lines such as:
+```
+T0.I  = _i:LINK(0).I
+```
 assign a temporary variable from a program-defined variable.   At this point the variable is assumed to be stacked at offset 0 on the stack.
 Offsets are relative to where the stack pointer is when entering the function.  The offset will be 0 at this point because register allocation and
```

Comment:
Upon review, of this comment, what line do you think should get a break? I was mostly trying to preserve the existing paragraph structure as that was easier to read through than code that's all on the same line.

Reply:
> I was mostly trying to preserve the existing paragraph structure as that was easier to read through than code that's all on the same line.

In this case you may just exactly do this with line breaks. As you are the current editor just do as it pleases you (as you've asked: I personally would add a break before "At this point" (because the previous half-sentence starts before the code sample) and "Offsets are relative" (for readability))